dandi/dandiapi.py

```diff
@@ -0,0 +1,348 @@
+        if result.status_code not in (200, 201):
+            raise HTTPError(
+                f"Error {result.status_code} while sending {method} request to {url}",
+                response=result,
+            )
+
+        if json_resp:
+            return result.json()
+        else:
+            return result
+
+    def get_url(self, path):
+        # Construct the url
+        if self.api_url.endswith("/") and path.startswith("/"):
+            path = path[1:]
+        url = self.api_url + path
+        return url
+
+    def get(self, path, parameters=None, json_resp=True):
+        """
+        Convenience method to call :py:func:`send_request` with the 'GET' HTTP method.
+        """
+        return self.send_request("GET", path, parameters, json_resp=json_resp)
+
+    def post(
+        self,
+        path,
+        parameters=None,
+        files=None,
+        data=None,
+        json=None,
+        headers=None,
+        json_resp=True,
+    ):
+        """
+        Convenience method to call :py:func:`send_request` with the 'POST' HTTP method.
+        """
+        return self.send_request(
+            "POST",
+            path,
+            parameters,
+            files=files,
+            data=data,
+            json=json,
+            headers=headers,
+            json_resp=json_resp,
+        )
+
+    def put(self, path, parameters=None, data=None, json=None, json_resp=True):
+        """
+        Convenience method to call :py:func:`send_request` with the 'PUT'
+        HTTP method.
+        """
+        return self.send_request(
+            "PUT", path, parameters, data=data, json=json, json_resp=json_resp
+        )
+
+    def delete(self, path, parameters=None, json_resp=True):
+        """
+        Convenience method to call :py:func:`send_request` with the 'DELETE' HTTP method.
+        """
+        return self.send_request("DELETE", path, parameters, json_resp=json_resp)
+
+    def patch(self, path, parameters=None, data=None, json=None, json_resp=True):
+        """
+        Convenience method to call :py:func:`send_request` with the 'PATCH' HTTP method.
+        """
+        return self.send_request(
+            "PATCH", path, parameters, data=data, json=json, json_resp=json_resp
+        )
+
+
+class DandiAPIClient(RESTFullAPIClient):
+    def get_asset(self, dandiset_id, version, uuid):
+        """
+
+        /dandisets/{version__dandiset__pk}/versions/{version__version}/assets/{uuid}/
+
+        Parameters
+        ----------
+        dandiset_id
+        version
+        uuid
+
+        Returns
+        -------
+
+        """
+        return self.get(f"/dandisets/{dandiset_id}/versions/{version}/assets/{uuid}/")
+
+    def get_dandiset(self, dandiset_id, version):
+        return self.get(f"/dandisets/{dandiset_id}/versions/{version}/")
+
+    def get_dandiset_assets(
+        self, dandiset_id, version, location=None, page_size=None, include_metadata=True
```

Comment:
I thought it would become "parametrized" for in the query for https://github.com/dandi/dandi-publish/issues/78 resolution, but now it always includes it, so indeed there is no need for it.  Removed in 515538eaccdba53d7f3f289c4e5e32f75b29e913